File: 0229-majority-element-ii/0229-majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
       
       int ele1 = 0;
       int cnt1 = 0;
       int ele2 = 0;
       int cnt2 = 0;
       int n = nums.size();
       for(int i=0;i<n;i++){
        if(cnt1 == 0 && ele2 != nums[i]){
            ele1 = nums[i];
            cnt1 = 1;
        }else if(cnt2 == 0 && ele1 != nums[i]){
            ele2 = nums[i];
            cnt2 = 1;
        }
        else if(nums[i] == ele1)cnt1++;
        else if(nums[i] == ele2)cnt2++;

        else{
            cnt1--, cnt2--;
        }
       }
      cnt1 =0, cnt2 =0;

      for(int i=0;i<n;i++){
        if(nums[i] == ele1)cnt1++;
        else if(nums[i] == ele2)cnt2++;
      }
      int mini = n/3;

      vector<int>ans;

      if(cnt1 > mini)ans.push_back(ele1);
      if(cnt2 > mini)ans.push_back(ele2);
      return ans;
    //   return {0};
    }
};

File: 0229-majority-element-ii/0229-majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        int len = nums.length;
        int ele1 = Integer.MIN_VALUE;
        int ele2 = Integer.MIN_VALUE;
        int c1 = 0;
        int c2 = 0;
        for(int i = 0; i < len; i++){
            if(c1 == 0 && nums[i] != ele2){
                ele1 = nums[i];
                c1=1;
            }else if(c2 == 0 && nums[i] != ele1){
                ele2 = nums[i];
                c2=1;
            }else if(nums[i] == ele1){
                c1++;
            }else if(nums[i] == ele2){
                c2++;
            }else{
                c1--;
                c2--;
            }
        }
        List<Integer>ans = new ArrayList<>();
        int cnt1 = 0;
        int cnt2 = 0;
        for(int i=0;i<len;i++){
            if(nums[i] == ele1)cnt1++;
            if(nums[i] == ele2)cnt2++;
        }
        if(cnt1 > Math.floor(len/3))ans.add(ele1);
        if(cnt2 > Math.floor(len/3))ans.add(ele2);
        return ans;
    }
}